xarray/tests/__init__.py

```diff
@@ -119,11 +118,10 @@ def assertItemsEqual(self, first, second, msg=None):
 
     @contextmanager
     def assertWarns(self, message):
-        with warnings.catch_warnings(record=True) as w:
-            warnings.filterwarnings('always', message)
+        with pytest.warns(Warning) as w:
             yield
-            assert len(w) > 0
-            assert any(message in str(wi.message) for wi in w)
+        assert len(w) > 0
```

Comment:
Yes, this was intentional. Python `with` blocks don't define scopes for variables, so `w` is still defined outside the block. In this case, the `with` block only indicates where warnings should be caught.